dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -299,18 +300,29 @@ private static void setMetadataRelatedIdentifiers(DepositMetadata depositMetadat
      * @param depositMetadata Metadata for the workflow version
      * @param workflow    workflow for which DOI is registered
      */
-    static void setMetadataCreator(DepositMetadata depositMetadata, Workflow workflow) {
-        final Stream<Author> authors = workflow.getAuthors().stream().map(ZenodoHelper::fromDockstoreAuthor);
-        final Stream<Author> orcidAuthors = workflow.getOrcidAuthors().stream()
+    static void setMetadataCreator(DepositMetadata depositMetadata, Workflow workflow, WorkflowVersion workflowVersion) {
+        // prefer authors from the specific workflow version
+        final Set<Author> setOfAuthors = new HashSet<>(getAuthors(workflowVersion.getAuthors(), workflowVersion.getOrcidAuthors()));
```

Comment:
This is good, it was probably a bug that, prior to this, we didn't prefer the version's authors.